hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/RMAppManager.java

```diff
@@ -469,26 +481,61 @@ private RMAppImpl createAndPopulateNewRMApp(
             submissionContext.getQueue() : placementContext.getFullQueuePath();
 
         String appName = submissionContext.getApplicationName();
-        CSQueue csqueue = ((CapacityScheduler) scheduler).getQueue(queueName);
+
+        CapacityScheduler cs = (CapacityScheduler) scheduler;
+        CapacitySchedulerConfiguration csConf = cs.getConfiguration();
+
+        CSQueue csqueue = cs.getQueue(queueName);
+        PrivilegedEntity privilegedEntity = getPrivilegedEntity(queueName);
 
         if (csqueue == null && placementContext != null) {
-          //could be an auto created queue through queue mapping. Validate
-          // parent queue exists and has valid acls
-          String parentQueueName = placementContext.getParentQueue();
-          csqueue = ((CapacityScheduler) scheduler).getQueue(parentQueueName);
+          List<Permission> permissions = new ArrayList<>();
```

Comment:
I think the whole CS ACL check logic could be extracted to somewhere else.

Reply:
Moved to CapacitySchedulerQueueManager.